crates/sparrow-instructions/src/evaluators/aggregation/token/collect_struct_token.rs

```diff
@@ -10,6 +13,14 @@ pub struct CollectStructToken {
     /// A [ListArray] comprised of lists of structs for each entity.
     #[serde(with = "sparrow_arrow::serde::array_ref")]
     pub state: ArrayRef,
+    /// Stores the times of the state values.
+    ///
+    /// A [ListArray] comprised of lists of timestamps for each entity.
+    ///
+    /// This array is only used when we have a `trailing` window.
+    /// Likely this should be separated into a different implementation.
```

Comment:
It's probably OK if we just leave it an empty array -- wouldn't take too much space.

Thoughts on whether it should be a `Vec<Vec<Timestamp>>` or something? I guess it could even be just a `Vec<Timestamp>` with flattening, but not sure which would be easiest to work with.

Reply:
Hm, definitely a possibility. Though, since the machinery exists for serde, it's easy to operate on and use `take` with, so will leave as is for this PR.